### ISSUE TITLE
Data too long for column 'referer'

### ISSUE THREAD

**Issue by itsmysterybox** [austintoddj/canvas#768]
My website is getting long referer urls. When I checked the log, this is what I found:

`[2020-06-29 14:09:28] production.ERROR: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'referer' at row 1 (SQL: insert into `canvas_views` (`post_id`, `ip`, `agent`, `referer`, `updated_at`, `created_at`) values (04259598-4807-43bd-9da3-d0702dc9f2ca, 139.195.125.194, Mozilla/5.0 (Linux; Android 9; SM-J610F) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Mobile Safari/537.36, https://www.novelupdates.com/reading-list/?__cf_chl_jschl_tk__=01a5641427511c7e1658da20a0204d555766d2cc-1587398236-0-AQRieKy_DPyUoaVjCEHKxehVZh9M9k58Qq48u7StLP3KujC_edcH5HgQdKXB1nIUcqlN4kZRY7t6APtwKe8vgeF3bA7TOrQ-jXkvSO8TSp3x9uQPp0KjjhMPfMA7BVFFAzHglJOo1Sdc2YJB3fG7TDaP0qa42vDKpkn5U8EgPeZThylQzRslUm76eKMRWMQN-EwNTa9xsjDDjExi4wbZT0ROX2OeYlRdU4Y3iho79-aknsTByUXEQJYb0Uqm1VDDdnG7qqSXkxN4o2ouBRuATC9DoY6_RVXfRg-t0fNBMHUa, 2020-06-29 14:09:28, 2020-06-29 14:09:28)) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'referer' at row 1 (SQL: insert into `canvas_views` (`post_id`, `ip`, `agent`, `referer`, `updated_at`, `created_at`) values (04259598-4807-43bd-9da3-d0702dc9f2ca, 139.195.125.194, Mozilla/5.0 (Linux; Android 9; SM-J610F) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.106 Mobile Safari/537.36, https://www.novelupdates.com/reading-list/?__cf_chl_jschl_tk__=01a5641427511c7e1658da20a0204d555766d2cc-1587398236-0-AQRieKy_DPyUoaVjCEHKxehVZh9M9k58Qq48u7StLP3KujC_edcH5HgQdKXB1nIUcqlN4kZRY7t6APtwKe8vgeF3bA7TOrQ-jXkvSO8TSp3x9uQPp0KjjhMPfMA7BVFFAzHglJOo1Sdc2YJB3fG7TDaP0qa42vDKpkn5U8EgPeZThylQzRslUm76eKMRWMQN-EwNTa9xsjDDjExi4wbZT0ROX2OeYlRdU4Y3iho79-aknsTByUXEQJYb0Uqm1VDDdnG7qqSXkxN4o2ouBRuATC9DoY6_RVXfRg-t0fNBMHUa, 2020-06-29 14:09:28, 2020-06-29 14:09:28)) at /var/www/konila.com/html/vendor/laravel/framework/src/Illuminate/Database/Connection.php:669)`

```
[previous exception] [object] (PDOException(code: 22001): SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'referer' at row 1 at /var/www/konila.com/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:127)
```

`referer` column is not accepting more than 255 characters. And the referers I'm getting have more than 400 characters. Possibly, the url can have more than 2000 characters, so `referer` column needs more bytes.

**Comment by austintoddj** [comment]
@itsmysterybox What database are you currently using?

**Comment by itsmysterybox** [comment]
I'm using MySQL.

**Comment by austintoddj** [comment]
I'm thinking that instead of modifying the database itself, I'd rather see the URLs cleaned up. If we used the `parse_url()` method in the `CaptureView` and `CaptureVisit` listeners, we would be able to avoid the above error. 

If you'd like to go ahead and give it a try, submit a PR to the `develop` branch and I'd be happy to help get it merged in.

**Comment by itsmysterybox** [comment]
Do you want to remove the query string from the referrer url? We don't really need that. And we are using only the host from `TopReferers` in stats, so scheme & host are important. And we may use path in future, I guess... What keys do you want to use for the referer?

**Comment by austintoddj** [comment]
Yeah, scheme and host are the 2 important parts. That should be all we need!

**Comment by austintoddj** [comment]
Fixed in [v6.0.0](https://github.com/austintoddj/canvas/releases/tag/v6.0.0)
